Client/Frontend/Browser/OpenInHelper.swift

```diff
@@ -24,38 +24,107 @@ struct OpenInViewUX {
 enum FileType : String {
     case PDF = "pdf"
     case PASS = "pkpass"
+    case HTML = "html"
+    case TXT = "txt"
+    case DEFAULT = "default"
+}
+
+enum MimeType: String {
+    case PDF = "application/pdf"
+    case PASS = "application/vnd.apple.pkpass"
+    case HTML = "text/html"
+    case TEXT = "text/plain"
+    case DEFAULT = "default"
 }
 
 protocol OpenInHelper {
-    var openInView: OpenInView { get }
-    static func canOpen(url: NSURL) -> Bool
+    init(url: NSURL)
+    init?(response: NSURLResponse)
```

Comment:
Nit: Looking below as to how both these `init` calls are being used, I think you can get away with using `init(url; NSURL)` as your designated initializer and making `init(response: NSURLResponse)` a convenience one that calls into `init(url..)`. Similar thing can be done for the `helperForURL`/`helperForResponse` in that you can make one a helper around the other since they both end up only using the URL anyways.